File: src/forge/service/main/java/settingdust/lazyyyyy/forge/service/util/LoaderAdapter.java
package settingdust.lazyyyyy.forge.service.util;

import net.minecraftforge.fml.loading.FMLLoader;
import net.minecraftforge.fml.loading.FMLPaths;
import net.minecraftforge.fml.loading.LoadingModList;

import java.nio.file.Path;

public class LoaderAdapter implements settingdust.lazyyyyy.util.LoaderAdapter {
    @Override
    public boolean isClient() {
        return FMLLoader.getDist().isClient();
    }

    @Override
    public boolean isModLoaded(String modId) {
        return LoadingModList.get().getModFileById(modId) != null;
    }

    @Override
    public Path getConfigDirectory() {
        return FMLPaths.CONFIGDIR.get();
    }

    @Override
    public Path getModsDirectory() {
        return FMLPaths.MODSDIR.get();
    }

    @Override
    public boolean hasEarlyErrors() {
        return !LoadingModList.get().getErrors().isEmpty();
    }
}

File: src/forge/game/main/kotlin/settingdust/lazyyyyy/forge/util/LoaderAdapter.kt
package settingdust.lazyyyyy.forge.util

import net.minecraftforge.fml.loading.FMLLoader
import net.minecraftforge.fml.loading.LoadingModList
import settingdust.lazyyyyy.util.LoaderAdapter

class LoaderAdapter : LoaderAdapter {
    override val isClient: Boolean
        get() = FMLLoader.getDist().isClient

    override fun isModLoaded(modId: String) = LoadingModList.get().getModFileById(modId) != null
}